server/game/gameSystems/DefeatCardSystem.ts

```diff
@@ -98,6 +94,8 @@ export class DefeatCardSystem<TContext extends AbilityContext = AbilityContext,
     }
 
     protected override updateEvent(event, card: Card, context: TContext, additionalProperties): void {
+        // if (card.zoneName !== ZoneName.Resource) {
```

Comment:
I think the commented out code here is actually what we should do - `addLeavesPlayPropertiesToEvent` shouldn't be called on defeated resources

Reply:
on OnCardDefeated event is empty if I uncomment these code

Reply:
Oh I know why. It's because `addLeavesPlayPropertiesToEvent` includes a call to `super.updateEvent` which is required. So we need to explicitly add that in the case where we don't call `addLeavesPlayPropertiesToEvent`